src/twisted/_version.py

```diff
@@ -7,5 +7,5 @@
 
 from incremental import Version
 
-__version__ = Version('Twisted', 17, 1, 0, dev=0)
+__version__ = Version('Twisted', 17, 1, 1, dev=0)
```

Comment:
This concern seems spurious to me, but I confess it might be a failure of imagination on my part.

Version constraints are for installing the correct version of a package from a package repository which contains releases.  In what circumstance would you find yourself with a random git revision of Twisted trunk as an installable option?  Pip implicitly excludes prereleases (including any `.devN`, I believe) if they were to happen to be uploaded to a package index; hence the need for the `--pre` flag.  You'd have to explicitly install a working copy of Twisted into an environment in order to trip this condition.  If you did that, you'd also find that pip doesn't include a constraint solver and will happily violate your previously-specified `==` and `>=` constraints.

If we let go of the idea that `.post0` is only for metadata changes, both of your proposed version specifiers still match, so I don't see how that would help.

So I can't imagine a scenario where `17.1.1.dev0` creates any practical problem; `17.1.0.dev0` creates a very real (frequent) annoyance with setuptools entrypoints, detailed on the incremental ticket.

That said: we could just say that the official release version to go on trunk is always just "always increment the minor release number, unless the previous release is 12, in which case increment the major and reset the minor to 1", as this would address your concern about pinning below a particular release, and even if we get the release prediction wrong, it doesn't cause any version specifiers to compare inaccurately.